src/modules/docker_context.rs

```diff
@@ -35,50 +42,57 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     )
     .join("config.json");
 
-    if !docker_config.exists() {
-        return None;
-    }
+    let docker_context_env = context
+        .get_env_os("DOCKER_HOST")
+        .or_else(|| context.get_env_os("DOCKER_CONTEXT"));
 
-    let json = utils::read_file(docker_config).ok()?;
-    let parsed_json = serde_json::from_str(&json).ok()?;
-
-    match parsed_json {
-        serde_json::Value::Object(root) => {
-            let current_context = root.get("currentContext")?;
-            match current_context {
-                serde_json::Value::String(ctx) => {
-                    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
-                        formatter
-                            .map_meta(|variable, _| match variable {
-                                "symbol" => Some(config.symbol),
-                                _ => None,
-                            })
-                            .map_style(|variable| match variable {
-                                "style" => Some(Ok(config.style)),
-                                _ => None,
-                            })
-                            .map(|variable| match variable {
-                                "context" => Some(Ok(ctx)),
-                                _ => None,
-                            })
-                            .parse(None)
-                    });
-
-                    module.set_segments(match parsed {
-                        Ok(segments) => segments,
-                        Err(error) => {
-                            log::warn!("Error in module `docker_context`:\n{}", error);
-                            return None;
-                        }
-                    });
-
-                    Some(module)
+    let ctx = match docker_context_env {
+        Some(data) => data,
+        _ => {
+            if !docker_config.exists() {
+                return None;
+            }
+            let json = utils::read_file(docker_config).ok()?;
+            let parsed_json = serde_json::from_str(&json).ok()?;
+
+            match parsed_json {
+                serde_json::Value::Object(root) => {
+                    let current_context = root.get("currentContext")?;
+                    match current_context {
+                        serde_json::Value::String(ctx_json) => OsString::from(ctx_json),
+                        _ => return None,
+                    }
                 }
```

Comment:
I think putting this into an `OsString` should be avoided.
I think it would be better to just use the non-`_os` version for getting the env var above or just call `.to_string_lossy()` and `.into_owned()` directly above.

And since you're touching this: You can simplify these lines here to `parsed_json.get("currentContext")?.as_str().map(str::to_owned)`